extras/ai-battle/main.cpp

```diff
@@ -0,0 +1,148 @@
+    {
+        bnw::cerr << "Error: " << e.what() << std::endl;
+        bnw::cerr << desc << std::endl;
+        return 1;
+    }
+
+    if(options.count("help"))
+    {
+        bnw::cout << desc << std::endl;
+        return 0;
+    }
+    if(options.count("version"))
+    {
+        bnw::cout << rttr::version::GetTitle() << " v" << rttr::version::GetVersion() << "-"
+                  << rttr::version::GetRevision() << std::endl
+                  << "Compiled with " << System::getCompilerName() << " for " << System::getOSName() << std::endl;
+        return 0;
+    }
+    if(options.count("map") == 0)
+    {
+        bnw::cerr << "No map specified" << std::endl;
+        return 1;
+    }
+    if(options.count("ai") == 0)
+    {
+        bnw::cerr << "No AI specified" << std::endl;
+        return 1;
+    }
+
+    try
+    {
+        auto random_init = static_cast<unsigned>(std::chrono::high_resolution_clock::now().time_since_epoch().count());
+        if(options.count("random_init"))
+            random_init = options["random_init"].as<unsigned>();
+
+        // We print arguments and seed in order to be able to reproduce crashes.
+        for(int i = 0; i < argc; ++i)
+            bnw::cout << argv[i] << " ";
+        bnw::cout << std::endl;
+        bnw::cout << "random_init: " << random_init << std::endl;
+        bnw::cout << std::endl;
+
+        RTTRCONFIG.Init();
+        RANDOM.Init(random_init);
+
+        const bfs::path mapPath = RTTRCONFIG.ExpandPath(options["map"].as<std::string>());
+        const std::vector<AI::Info> ais = ParseAIOptions(options["ai"].as<std::vector<std::string>>());
+
+        GlobalGameSettings ggs;
+        if(options.count("objective"))
+        {
+            std::string objective = options["objective"].as<std::string>();
+            if(objective == "domination")
+                ggs.objective = GameObjective::TotalDomination;
+            else if(objective == "conquer")
+                ggs.objective = GameObjective::Conquer3_4;
+            else
+            {
+                bnw::cerr << "unknown objective: " << objective << std::endl;
+                return 1;
+            }
+        } else
+            ggs.objective = GameObjective::TotalDomination;
+
+        ggs.objective = GameObjective::TotalDomination;
+        HeadlessGame game(ggs, mapPath, ais);
+        if(options.count("replay"))
+            game.StartReplay(options["replay"].as<std::string>(), random_init);
+
+        unsigned maxGF = std::numeric_limits<unsigned>::max();
+        if(options.count("maxGF"))
+            maxGF = options["maxGF"].as<unsigned>();
+
+        game.Run(maxGF);
+        game.Close();
+        if(options.count("save"))
+            game.SaveGame(options["save"].as<std::string>());
+    } catch(const std::exception& e)
+    {
+        bnw::cerr << e.what() << std::endl;
+        return 1;
+    } catch(...)
+    {
+        bnw::cerr << "An unknown exception occurred" << std::endl;
+        return 1;
+    }
```

Comment:
Optional:
Why catch it at all? I'd think the error message by the system without this catch should have more information about this unknown error
```suggestion
    }
```